src/meta.rs

```diff
@@ -34,32 +38,150 @@ impl ResourceMeta {
+        // fn is_plain(_: &[u8]) -> bool {
+        //     true
+        // };
+        // let g = infer::get_from_path(path.as_path()).unwrap().unwrap_or(infer::Type::new(infer::MatcherType::Text, "text/plain", "txt", is_plain))
+        // match g.matcher_type() {
+        //     MatcherType::Image => {
+        //         match g.mime_type() {
+        //             "image/jpeg"| "image/jpg"| "image/png"| "image/gif" => {
+
+        //             },
+        //             _ => {}
+        //         }
+        //     }
+        //     MatcherType::Archive => {
+        //         match g.mime_type() {
+
+        //         }
+        //     }
+        //     _ => {}
+        // }
+        let ext = path
+            .extension()
+            .unwrap_or_default()
+            .to_str()
+            .unwrap_or_default();
+
+        if ext == "link" {
+            let link =  match parse_link(&path) {
+                Ok(x) => x,
+                Err(e) => {
+                    log::error!(
+                        "cannot parse link: {}, fallback to default plaintext factory",
+                        path.display()
+                    );
+                    Self::PlainText
+                }
+            };
+            return link;
+        }
+
+        let accepted_doc = ["pdf", "doc", "docx", "odt", "ods", "md"];
+        let accepted_img = ["jpg", "jpeg", "png", "svg", "gif"];
+        let accepted_ar = ["zip", "7z", "rar", "tar.gz", "tar.xz"];
+        let generic_text = ["txt", ""];
+        let accepted_video = [
+            "mp4", "avi", "mkv", "mov", "wmv", "flv", "webm", "ts", "mpg",
+        ];
+        if accepted_ar.contains(&ext) {
+            return ResourceKind::Archive;
+        };
+        if accepted_img.contains(&ext) {
+            return ResourceKind::Image;
+        }
+        if accepted_doc.contains(&ext) {
+            if ext != "pdf" {
+                return ResourceKind::Document { pages: None };
+            } else {
+                // TODO: Wait for
+                let pb = pdf::initialize_pdfium();
+                let pages = Some(
+                    Pdfium::new(pb)
+                        .load_pdf_from_file(path.as_path(), None)
+                        .unwrap()
+                        .pages()
+                        .len() as i64,
+                );
+
+                return ResourceKind::Document { pages };
+            }
+        }
+
+        if accepted_video.contains(&ext) {
+            // TODO: Read Video Info
+            return ResourceKind::PlainText;
+        }
+        ResourceKind::PlainText
+    }
+}
 
 fn convert_str(option: Option<&OsStr>) -> Option<OsString> {
     if let Some(value) = option {
         return Some(value.to_os_string());
     }
     None
 }
+#[derive(Deserialize, Serialize)]
+pub struct LinkFile {
+    pub title: String,
+    pub desc: String,
+    pub url: String,
+}
+
+fn parse_link(path: &CanonicalPathBuf) -> Result<ResourceKind, Error> {
```

Comment:
We have Link-to-Web in `arklib` now, pull latest commits from `main`.

Reply:
Got. Will merge it later.